packages/contracts-bedrock/src/L1/StandardValidator.sol

```diff
@@ -0,0 +1,520 @@
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+import { IOptimismMintableERC20Factory } from "interfaces/universal/IOptimismMintableERC20Factory.sol";
+import { IL1StandardBridge } from "interfaces/L1/IL1StandardBridge.sol";
+import { IL1ERC721Bridge } from "interfaces/L1/IL1ERC721Bridge.sol";
+import { IPermissionedDisputeGame } from "interfaces/dispute/IPermissionedDisputeGame.sol";
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+import { IDelayedWETH } from "interfaces/dispute/IDelayedWETH.sol";
+import { ISemver } from "interfaces/universal/ISemver.sol";
+import { IResourceMetering } from "interfaces/L1/IResourceMetering.sol";
+import { IOptimismPortal2 } from "interfaces/L1/IOptimismPortal2.sol";
+import { IPreimageOracle } from "interfaces/cannon/IPreimageOracle.sol";
+
+contract StandardValidatorV180 {
+    Implementations internal impls;
+    ISuperchainConfig internal superchainConfig;
+    address internal l1PAOMultisig;
+    address internal mips;
+    address internal guardian;
+    address internal challenger;
+
+    // Semver strings for our contracts. Pulled from the Superchain Registry.
+    string internal constant systemConfigVersion = "2.3.0";
+    string internal constant permissionedDisputeGameVersion = "1.3.1";
+    string internal constant mipsVersion = "1.2.1";
+    string internal constant optimismPortalVersion = "3.10.0";
+    string internal constant anchorStateRegistryVersion = "2.0.0";
+    string internal constant delayedWETHVersion = "1.1.0";
+    string internal constant disputeGameFactoryVersion = "1.0.0";
+    string internal constant l1CrossDomainMessengerVersion = "2.3.0";
+    string internal constant l1ERC721BridgeVersion = "2.1.0";
+    string internal constant l1StandardBridgeVersion = "2.1.0";
+    string internal constant optimismMintableERC20FactoryVersion = "1.9.0";
+    string internal constant preimageOracleVersion = "1.1.2";
+
+    struct Implementations {
+        address l1ERC721BridgeImpl;
+        address optimismPortalImpl;
+        address systemConfigImpl;
+        address optimismMintableERC20FactoryImpl;
+        address l1CrossDomainMessengerImpl;
+        address l1StandardBridgeImpl;
+        address disputeGameFactoryImpl;
+        address anchorStateRegistryImpl;
+        address delayedWETHImpl;
+        address mipsImpl;
+    }
+
+    struct Input {
+        IProxyAdmin proxyAdmin;
+        ISystemConfig sysCfg;
+        bytes32 absolutePrestate;
+        uint256 l2ChainID;
+    }
+
+    constructor(
+        Implementations memory _implementations,
+        ISuperchainConfig _superchainConfig,
+        address _l1PAOMultisig,
+        address _mips,
+        address _guardian,
+        address _challenger
+    ) {
+        impls = _implementations;
+        superchainConfig = _superchainConfig;
+        l1PAOMultisig = _l1PAOMultisig;
+        mips = _mips;
+        guardian = _guardian;
```

Comment:
We can remove guardian as an input from the constructor and read it from the superchain config